account/account_test.go

```diff
@@ -351,3 +355,145 @@ func TestConvertGDPREnforcePurposeFields(t *testing.T) {
 		}
 	}
 }
+
+func TestGdprCcpaChannelEnabled(t *testing.T) {
+	testCases := []struct {
+		description         string
+		givenAccountID      string
+		givenMetric         string
+		required            bool
+		disabled            bool
+		expectedMetricCount int
+	}{
+		{
+			description:         "Account is GDPR Channel Enabled, the GDPR metric should be called",
+			givenAccountID:      "gdpr_channel_enabled_acct",
+			givenMetric:         "RecordAccountGDPRChannelEnabledWarning",
+			required:            false,
+			disabled:            false,
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Account is CCPA Channel Enabled, the CCPA metric should be called",
+			givenAccountID:      "ccpa_channel_enabled_acct",
+			givenMetric:         "RecordAccountCCPAChannelEnabledWarning",
+			required:            false,
+			disabled:            false,
+			expectedMetricCount: 1,
+		},
+		{
+			description:         "Account isn't channel enabled, CCPA metric shouldn't be called",
+			givenAccountID:      "valid_acct",
+			givenMetric:         "RecordAccountCCPAChannelEnabledWarning",
+			required:            false,
+			disabled:            false,
+			expectedMetricCount: 0,
+		},
+		{
+			description:         "Account isn't channel enabled, GDPR metric shouldn't be called",
+			givenAccountID:      "valid_acct",
+			givenMetric:         "RecordAccountGDPRChannelEnabledWarning",
+			required:            false,
+			disabled:            false,
+			expectedMetricCount: 0,
+		},
+	}
+
+	for _, test := range testCases {
```

Comment:
You probably want to add a name field with a much smaller name with no spaces instead of `test.description`. Makes the tests easier to reference, and save the test framework from having to modify the name to not have spaces.